proposals/compatibility-mode.md

```diff
@@ -95,6 +95,17 @@ Use of the `sample_mask` builtin would cause a validation error at pipeline crea
 
 **Justification**: OpenGL ES 3.1 does not support `gl_SampleMask` or `gl_SampleMaskIn`.
 
+### 8. Disallow `linear` and `sample` interpolation options.
+
+In WGSL, an inter-stage variable can be marked with one of three interpolation types: `'perspective'`,
+`'linear'`, `'flat'`, and one of three interpolation sampling modes: `'center'`, `'centroid'`, `'sample'`
+
+In compatibility mode, `'linear'` type and sampling mode `'sample'` are disallowed.
+If used in a shader module passed to `createRenderPipeline`, `createRenderPipelineAsync`,
+`createComputePipeline`, or `createComputePipelineAsync` a validation error is generated.
```

Comment:
I think that for things that become unavailable in WGSL we should only raise errors if they are used by the entry point.
This will allow uses of those items in modules that are meant to run on compat and non-compat and is also consistent with our existing "statically accessed" rules in the "12.3. Shader Interface" section.

Reply:
Agreed. That was the intent. I guess it should be spelled out. I'll update